File: Java/LV3/최고의 집합.java
import java.util.*;

class Solution {
    public int[] solution(int n, int s) {
        int[] answer = {};
        
        if(s/n<1){
            answer = new int[1];
            answer[0]=-1;
        }
        
        else{
            int partition = s/n;
            int remainder = s%n;
            int counter = 0;
            answer = new int[n];
            for(int i=0;i<remainder;i++){
                answer[i]=partition+1;
                counter++;
            }
            
            for(int i=counter;i<n;i++){
                answer[i]=partition;
            }
        }
        
        Arrays.sort(answer);
        return answer;
    }
}

File: Lv.3/최고의 집합.cpp
#include <string>
#include <vector>
#include <iostream>
#include <functional>
#include <algorithm>
#include <limits.h>
using namespace std;

// 아이디어 참신함!
vector<int> solution(int n, int s) {
    vector<int> answer;
    
    if(s/n<1){
        answer.push_back(-1);
    }
    
    else{
        
        int partition = s/n;
        int remainder = s%n;
        int counter = 0;
        
        for(int i=0;i<remainder;i++){
            answer.push_back(partition+1);
            counter++;
        }
        
        for(int i=counter+1;i<=n;i++){
            answer.push_back(partition);
        } 
        
    }
    
    sort(answer.begin(),answer.end());
    return answer;
}


// 11 3 - partition:3  remainder:2
// 3 4 4
    
    
